src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -101,8 +112,11 @@ public KNNQueryBuilder(StreamInput in) throws IOException {
             fieldName = in.readString();
             vector = in.readFloatArray();
             k = in.readInt();
+            if (isClusterOnOrAfterMinRequiredVersion()) {
```

Comment:
Is there a comment somewhere describing what we are doing here?

Reply:
Nope, there isn't one, let me add it